src/Http/Middleware/ProxyAddress.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Http\Middleware;
+
+use Flarum\Http\Exception\ProxyNotAllowedException;
+use Illuminate\Support\Arr;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface as Middleware;
+use Psr\Http\Server\RequestHandlerInterface;
+
+class ProxyAddress implements Middleware
+{
+    /**
+     * @var bool
+     */
+    protected $enabled;
+
+    /**
+     * @var array
+     */
+    protected $allowedAddresses;
+
+    /**
+     * @param bool $enabled
+     * @param array $allowedAddresses
+     */
+    public function __construct($enabled, $allowedAddresses)
+    {
+        $this->enabled = $enabled;
+        $this->allowedAddresses = $allowedAddresses;
+    }
+
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        $ipAddress = Arr::get($request->getServerParams(), 'REMOTE_ADDR', '127.0.0.1');
+
+        if ($this->enabled) {
+            if (in_array($ipAddress, $this->allowedAddresses)) {
```

Comment:
Perhaps we should support wildcards here?

Reply:
I actually agree, wildcards would be great, even better if we could find a way to support CIDR notation easily as something like `169.254.56.*` might include multiple ISPs, services, or companies. A CIDR is much more accurate.

(Note: Example IP is non-routable and safe for example use)